packages/@react-stately/tag/src/useTagGroupState.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {TagGroupProps, TagGroupState} from '@react-types/tag';
+import {useEffect, useRef} from 'react';
+import {useListState} from '@react-stately/list';
+
+/**
+ * Provides state management for a TagGroup component.
+ */
+export function useTagGroupState<T extends object>(props: TagGroupProps<T>): TagGroupState<T> {
+  let state = useListState(props);
+
+  let keyToRestoreOnRemove = useRef(null);
+
+  useEffect(() => {
+    // If the focused key is removed, restore focus to the tag after, or tag before if no tag after.
+    if (!state.collection.getItem(state.selectionManager.focusedKey)) {
```

Comment:
I don't think this is quite right, this will happen for reasons other than the Tag remove button being called

I think useTagGroupState should have a callback 'onRemove' that the aria implementation can call when a tag is removed. Then here we can manage focus at that point in time. It'll be before the next render that way as well, instead of being a render behind as it is now

if a tag is removed some other way, then it is likely that focus isn't on the taggroup at that time, so it's ok I think for now if focus is lost in those cases to the collection of tags

Reply:
I thought that might be the case. Just updated based on this recommendation, thanks.